sdk/scheduler/src/test/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluatorLinuxCapabilitiesTest.java

```diff
@@ -0,0 +1,146 @@
+                        UUID.randomUUID(),
+                        ArtifactResource.getUrlFactory(TestConstants.SERVICE_NAME),
+                        SchedulerConfigTestUtils.getTestSchedulerConfig(),
+                        Collections.emptyList(),
+                        TestConstants.FRAMEWORK_ID,
+                        Collections.emptyMap()));
+
+
+        Assert.isTrue(outcome.isPassing());
+    }
+
+    @Test
+    public void testOfferNoCapabilites() throws Exception {
+
+        String agent = "test-agent";
+        Protos.Resource offered = ResourceTestUtils.getUnreservedCpus(1.0);
+        Protos.Offer offer = offerWithAgent(agent, offered);
+
+        MesosResourcePool mesosResourcePool = new MesosResourcePool(offer, Optional.of(Constants.ANY_ROLE));
+
+        LinuxCapabilitiesEvaluationStage capabilityEvaluationStage = new LinuxCapabilitiesEvaluationStage(TestConstants.TASK_NAME, Optional.empty());
+        PodSpec podSpec = PodInstanceRequirementTestUtils.getCpuRequirement(1.0).getPodInstance().getPod();
+        PodInstance podInstance = new DefaultPodInstance(podSpec, 0);
+        List<String> taskNames = TaskUtils.getTaskNames(podInstance);
+
+        podInstance.getPod().getCapabilities().clear();
+
+        PodInstanceRequirement podInstanceRequirement = PodInstanceRequirement
+                .newBuilder(podInstance, taskNames)
+                .build();
+
+        EvaluationOutcome outcome = capabilityEvaluationStage.evaluate(
+                mesosResourcePool,
+                new PodInfoBuilder(
+                        podInstanceRequirement,
+                        TestConstants.SERVICE_NAME,
+                        UUID.randomUUID(),
+                        ArtifactResource.getUrlFactory(TestConstants.SERVICE_NAME),
+                        SchedulerConfigTestUtils.getTestSchedulerConfig(),
+                        Collections.emptyList(),
+                        TestConstants.FRAMEWORK_ID,
+                        Collections.emptyMap()));
+
+
+        Assert.isTrue(outcome.isPassing());
+    }
+
+    @Test
+    public void testOfferPartialCapabilites() throws Exception {
+
+        String agent = "test-agent";
+        Protos.Resource offered = ResourceTestUtils.getUnreservedCpus(1.0);
+        Protos.Offer offer = offerWithAgent(agent, offered);
+
+        MesosResourcePool mesosResourcePool = new MesosResourcePool(offer, Optional.of(Constants.ANY_ROLE));
+
+        LinuxCapabilitiesEvaluationStage capabilityEvaluationStage = new LinuxCapabilitiesEvaluationStage(TestConstants.TASK_NAME, Optional.empty());
+        PodSpec podSpec = PodInstanceRequirementTestUtils.getCpuRequirement(1.0).getPodInstance().getPod();
+        PodInstance podInstance = new DefaultPodInstance(podSpec, 0);
+        List<String> taskNames = TaskUtils.getTaskNames(podInstance);
+
+        podInstance.getPod().getCapabilities().clear();
+
+        //TODO: how to alter the podspec in a test?
```

Comment:
Use `DefaultPodSpec.newBuilder(PodSpec)`. Then you can easily alter the PodSpec.